app/javascript/components/zone-form/index.jsx

```diff
@@ -66,7 +66,7 @@ const ZoneForm = ({ recordId }) => {
 };
 
 ZoneForm.propTypes = {
-  recordId: PropTypes.string,
+  recordId: PropTypes.number,
```

Comment:
i think this should be string, in zone edit form, recordId should be passed as string, you can add to_s.